lib/tree.js

```diff
@@ -7,18 +7,36 @@ var metadata = require('./helpers/metadata');
 function tree(opts) {
   opts = _.defaults(opts || {}, {
     deep: false,
+    depth: null,
   });
 
   var tasks = this._registry.tasks();
-  var nodes = _.map(tasks, function(task) {
-    var meta = metadata.get(task);
 
-    if (opts.deep) {
-      return meta.tree;
-    }
-
-    return meta.tree.label;
-  });
+  var nodes;
+  if (!opts.deep) {
```

Comment:
can you normalize `opts.deep` to `depth: Infinity` and avoid the branching logic?  Deep can be there for backwards compat or we can just remove it and release as a breaking change (this module isn't 1.0 yet)